sirepo/pkcli/service.py

```diff
@@ -52,25 +59,159 @@ def flower():
         ])
 
 
-def http():
-    """Starts Flask server in http mode.
+def flask():
+    from sirepo import server
+
+    use_reloader = pkconfig.channel_in('dev')
+    app = server.init(use_reloader=use_reloader)
+    # avoid WARNING: Do not use the development server in a production environment.
+    app.env = 'development'
+    app.run(
+        host=cfg.ip,
+        port=cfg.port,
+        threaded=True,
+        use_reloader=use_reloader,
+    )
+
+
+def http(driver='local', nersc_proxy=None, nersc_user=None, sbatch_host=None):
+    """Starts the Flask server and job_supervisor.
 
     Used for development only.
+
+    Args:
+        driver (string): The driver type to enable (one of local, docker, sbatch, nersc)
+        nersc_proxy (string): A proxy nersc can use to reach the supervisor
+        nersc_user(string): A nersc user ??
+        sbatch_host (string): A host to ssh into to start sbatch jobs
     """
-    from sirepo import server
+    def _env(py_version):
+        e = os.environ.copy()
+        e.update(
+            PYENV_VERSION=py_version,
+            PYKERN_PKDEBUG_WANT_PID_TIME='1',
+            SIREPO_AUTH_EMAIL_FROM_EMAIL='support@radiasoft.net',
+            SIREPO_AUTH_EMAIL_FROM_NAME='RadiaSoft Support',
+            SIREPO_AUTH_EMAIL_SMTP_PASSWORD='n/a',
+            # POSIT: same as sirepo.auth.email._DEV_SMTP_SERVER
+            SIREPO_AUTH_EMAIL_SMTP_SERVER='dev',
+            SIREPO_AUTH_EMAIL_SMTP_USER='n/a',
+            SIREPO_AUTH_METHODS='email:guest',
+            SIREPO_MPI_CORES='2',
+        )
+        {
+            n: f for n, f in inspect.currentframe().f_back.f_locals.items()
+            if callable(f)
+        }['_env_' + driver](e)
+
+        if 'sbatch' in e['SIREPO_JOB_DRIVER_MODULES']:
+            h = e['SIREPO_JOB_DRIVER_SBATCH_HOST']
+            m = re.search(r'^{}.+$'.format(h), pkio.read_text('~/.ssh/known_hosts'), re.MULTILINE)
+            assert m, \
+                (
+                    'you need to get the host key in ~/.ssh/known_hosts'
+                    ' run: `ssh {} true`'.format(h)
+                )
+            x = m.group(0)
+            e.update(SIREPO_JOB_DRIVER_SBATCH_HOST_KEY=x)
+        return e
+
+    def _env_docker(env):
+        # TODO(e-carlin): py3 has shutil.which()
```

Comment:
There are some differences between py2 and py3 for finding if commands exist or not. How do you want to solve this? Wrap it? Open an issue and link it to the py3 conversion work?